app/models/results/blank_response_tree_builder.rb

```diff
@@ -3,37 +3,41 @@
 module Results
   # Builds and saves response tree, with all blank answers, from a form object.
   class BlankResponseTreeBuilder
-    attr_accessor :response
+    attr_accessor :response, :options
 
     delegate :form, to: :response
 
-    def initialize(response)
+    def initialize(response, options = {})
       self.response = response
+      self.options = options
     end
 
     def build
-      root = AnswerGroup.create!(form_item: form.root_group, response: response)
+      root = AnswerGroup.new(form_item: form.root_group, response: response)
+      root.save! if options[:save]
       add_level(form.root_group.sorted_children, root)
       response.associate_tree(root)
 
       # TODO: We can remove the `validate: false` once various validations are
       # removed from the response model
-      response.save(validate: false)
+      response.save(validate: false) if options[:save]
```

Comment:
I think you can get rid of save altogether since we gave up on that approach.

You just need to change one thing to keep specs from breaking:

```
-      root = AnswerGroup.create!(form_item: form.root_group, response: response)
+      root = AnswerGroup.create!(form_item: form.root_group, response: response, new_rank: 0)
```